### DISCUSSION TITLE
[Detector Support]: Possible to train the detector in 0.17 Beta (non Frigate+) to differentiate between a car and a person now?

### DISCUSSION THREAD

**Discussion by sharpcall** [blakeblackshear/frigate#21425] [Detector Support]
### Describe the problem you are having

Is there a way to tell Frigate that it is seeing a car currently and not a person in Frigate 0.17 Beta (not Frigate+)? Is it trainable at all?

### Version

0.17 beta

### Frigate config file

```yaml
NA
```

### docker-compose file or Docker CLI command

```yaml
NA
```

### Relevant Frigate log output

```shell
NA
```

### Install method

Docker Compose

### Object Detector

Coral

### Screenshots of the Frigate UI's System metrics pages

NA

**Comment by inkeep-ai-assistant** [comment]
Based on the available documentation, there isn't a built-in way to manually train or correct the base object detector in Frigate 0.17 Beta (non-Frigate+) to differentiate between a car and a person[(1)](https://docs.frigate.video/configuration/object_detectors).

The base Frigate installation uses pre-trained models that detect objects from a fixed set of labels[(2)](https://docs.frigate.video/configuration/objects). The default model includes both "person" and "car" as tracked objects[(2)](https://docs.frigate.video/configuration/objects). However, if the detector is misidentifying objects, you cannot directly train or fine-tune the base model without Frigate+[(1)](https://docs.frigate.video/configuration/object_detectors).

**Your options without Frigate+:**

1. **Adjust detection parameters** - You can tune `min_score` and `threshold` values in your config to filter out low-confidence detections, though this won't fix misclassification issues[(3)](https://docs.frigate.video/configuration/review).

2. **Use zones and masks** - Configure zones to restrict where detections are considered valid, or use masks to exclude problematic areas[(3)](https://docs.frigate.video/configuration/review).

3. **Try different detector models** - Frigate supports multiple detector types (EdgeTPU, OpenVINO, ONNX, etc.) and different model architectures that may perform better for your specific scenario[(1)](https://docs.frigate.video/configuration/object_detectors).

**With Frigate+:**

Frigate+ provides community-trained models with significantly improved accuracy and includes a fine-tuning workflow where you can upload images and correct false positives/negatives to create scene-specific models[(4)](https://github.com/blakeblackshear/frigate/discussions/9098). The Frigate+ models are specifically trained on real security camera footage and perform much better at distinguishing between objects, especially in challenging conditions like nighttime/IR[(4)](https://github.com/blakeblackshear/frigate/discussions/9098).

If you're experiencing persistent misclassification issues with the base model, subscribing to Frigate+ would be the recommended solution for training the detector to better differentiate between cars and people in your specific environment.

📚 **Sources:**
1. [Supported Hardware](https://docs.frigate.video/configuration/object_detectors)
2. [Available Objects](https://docs.frigate.video/configuration/objects)
3. [Review](https://docs.frigate.video/configuration/review)
4. [Training Frigate+ for Vehicle detection in Black & White / IR at night](https://github.com/blakeblackshear/frigate/discussions/9098)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by NickM-27** [comment]
This has always been possible, but not within Frigate. Training an object detection model is significantly more complicated and resource intensive than a classification model.

**Reply by haldi4803** [reply]
Is anyone aware of an Docker/Software that accesses Frigate via MQTT to automatically get the necessary files for training a lora (or whatever) and has all that included in a neat web interface?

**Reply by hawkeye217** [reply]
No, I don't think anything like this is available. [Frigate+](https://frigate.video/plus/) is the only integrated option today that provides a hosted annotation and training workflow.

**Reply by sharpcall** [reply]
so the annual subscription is needed for training twice a year for free?
Is there a need to train more than twice? is it related to camera position?

**Reply by hawkeye217** [reply]
12 model trainings per year are included. See the official docs: https://docs.frigate.video/plus/

**Comment by hawkeye217** [comment]
See the [FAQ](https://github.com/blakeblackshear/frigate/discussions/21275), Usage question 5.
